### ISSUE TITLE
Various update-cdk task issues: cdk destroy invocation, CDK CLI prompt hangs

### ISSUE THREAD

**Issue by rob3c** [org-formation/org-formation-cli#313]
As requested, first I'll just copy the original message from the org-formation Slack thread in `#general` [here](https://org-formation.slack.com/archives/CT6JQ05AA/p1643943065291739):

 Hi, I have some questions about using the AWS CDK with org-formation:

1. Is anyone actually using the `update-cdk` task as-is, or do we really need to use custom commands for realistic scenarios? For example, many deployments get caught by confirmation prompts in the CDK CLI, but `update-cdk` doesn't use or have flags for either `--require-approval never` for `cdk deploy` or `--force` for `cdk destroy`.
2. Speaking of `cdk destroy`, does that ever really get invoked? I see the `performRemove()` function in the cdk plugin source code that builds a `cdk destroy` command, but I can't seem to trigger either it or a `CustomRemoveCommand` replacement for it. If I remove an `update-cdk` task, nothing happens. If I change the org bindings for it, any new accounts in the binding get deployed, but any accounts removed from the binding are just ignored.
3. Since standard use of the `update-cdk` task doesn't seem to invoke `cdk destroy` or the custom equivalent, I've just been explicitly calling `cdk destroy` by changing the `CustomDeployCommand` to use it in stead of `cdk deploy` in order to actually remove CDK tasks. Any changes to bindings get reshuffled between multiple `update-cdk` org projections - some for the deployments, and the others for the deletions. Is anyone else doing this, or am I missing something?
4. Given the above issues using the preferred-but-problematic org-formation declarative CloudFormation syntax with `update-cdk`, is anyone skipping this altogether and tried using the [org-formation-client](https://github.com/org-formation/org-formation-client) project within a CDK app to directly query for deployment accounts from the other direction? I've considered this but have been avoiding it, but maybe this way could be easier to sync CDK stacks with org changes (combined with listening for org events on the event bus)?

Those are the main issues. I'm overall happy with org-formation, and I'm just looking to see if there's a more ergonomic way to use the CDK with it that I'm missing. Thanks in advance for your consideration.

**Comment by rob3c** [comment]
So now that I've copied the issue over to get started, I think this boils down to two main issues:

1. Avoid hangs during automated builds due to the AWS CDK CLI confirmation prompts.

Unfortunately, the way you avoid them is different between `cdk deploy` and `cdk destroy`, but they both involve using certain flags:

- For `cdk deploy`, use `--require-approval never` to skip the change confirmation prompt.
- For `cdk destroy`, use `--force` to skip the delete confirmation prompt.

I think for likely use-cases of org-formation in automation, these should be the default behavior. These flags for forcing deploy/destroy without prompting are semantically different than the `Force: True` property on tasks, so we should probably add separate property flags to the `update-cdk` task. Maybe something like `EnableConfirmationPrompts` that suggests that the default value is False but can be overridden in special circumstances.

2. Ensure that `cdk destroy` is called when an `update-cdk` task is removed or some of its organization bindings have been removed. The CDK is sensitive to being called with the same parameters when destroying stacks as was called when deploying them. However, CDK users should already be used to dealing with this, so we can probably just make sure that the remove command is invoked where expected without worrying about using parameters from the previous state vs the current task params.

**Comment by rehos** [comment]
For what's it worth, to avoid the use of the command line options like `--require-approval` and `--force` you can also modify the cdk.json file and add:

```JSON
{
    ...
    "requireApproval": "never",
    "force": true,
    ...
}
```
> Disclaimer: I haven't run any delete tasks lately, but I assume `force` is still working

**Comment by rob3c** [comment]
Yes, modifying cdk.json is another way to skip the confirmations besides using custom commands. However, then local dev usage where you probably want the confirmations (unlike CI/CD pipelines) requires modifying the file locally (hopefully, not committing it by accident) or manually adding flags to override them. I don't see that as helping devs fall into the pit of success, but it is another working alternative.
